src/test/java/com/google/cloud/spanner/jdbc/MultiplexedSessionsMockServerTest.java

```diff
@@ -102,6 +102,38 @@ public void testUsesMultiplexedSessionForQueryInAutoCommit() throws SQLException
     assertTrue(session.getMultiplexed());
   }
 
+  @Test
+  public void testUsesMultiplexedSessionForQueryInReadOnlyTransaction() throws SQLException {
+    int numQueries = 2;
+    try (Connection connection = createConnection()) {
```

Comment:
nit: do we need to explicitly call connection#commit() or does it automatically happen when it comes out of try block?

Reply:
Closing a connection with an active transaction will automatically commit that transaction, so in this case we could skip the `commit()` call. It is however not something I would consider good practice, as letting a transaction be implicitly committed makes it harder to read/understand the code.

I would for example normally assume that an active transaction would be rolled back if the connection is closed while the transaction is active, but the JDBC spec requires the driver implementation to commit the transaction. This might not be known to everyone, which again could lead people to believe that the transaction is rolled back instead of committed.